.github/workflows/CI-Batched.yml

```diff
@@ -664,7 +664,7 @@ jobs:
 
   validate-all-tests-pass:
     runs-on: ubuntu-latest
-    needs: [run-batch-job,e2etest-preparation]
+    needs: [run-batch-job,e2etest-preparation,create-test-ref]
```

Comment:
Looks like we already have this [need](https://github.com/aws-observability/aws-otel-collector/blob/main/.github/workflows/CI-Batched.yml#L582) for `run-batch-job` and since run-batch-job is already a need for this step. is this a required change.